판매자 페이지 UI 구현 및 메인 홈 반응형 개선

1000hyehyang opened the pull request:
## ⭐️ Issue Number

- #60

## 🚩 Summary

- 판매자 상품 등록 및 공구 등록 페이지 UI 구현 (판매자 페이지는 반응형 구현 X -> 해당 페이지는 모바일/태블릿 사용 가능성 낮음)
![image](https://github.com/user-attachments/assets/d88aa51d-69a0-4ce4-8a27-47284c057ec9)
![image](https://github.com/user-attachments/assets/337a82ab-6cd2-4390-9f7f-19262e9af4ea)


- 메인 홈페이지 1280px 이상 화면에서 그리드 레이아웃 깨짐 현상 수정
![image](https://github.com/user-attachments/assets/1f1f37bd-e29d-4f46-815f-73b2ea32588b)

- 이벤트 페이지와 숏구 페이지 추가 및 모든 페이지 연결 완료 (준비중 페이지로)
- 전체 페이지의 최대 너비를 1248px에서 1280px로 통일하여 일관성 확보
- 데스크탑 그리드 레이아웃 최적화 (2xl 브레이크포인트 제거)

## 🛠️ Technical Concerns


### 반응형 레이아웃 최적화

- **브레이크포인트 전략 재검토**: `2xl` 브레이크포인트 제거로 1280px 이상 화면에서 그리드 깨짐 현상 해결
- **CSS Grid 시스템 개선**:
  ```css
  /* Before: lg:grid-cols-4 2xl:grid-cols-6 */
  /* After: xl:grid-cols-4 */
  ```
- **컨테이너 너비 표준화**: 모든 페이지의 최대 너비를 1280px로 통일하여 일관된 사용자 경험 제공

### 컴포넌트 아키텍처 개선

- **재사용 가능한 EmptyPage 컴포넌트**:
  ```typescript
  interface EmptyPageProps {
    title?: string;
    description?: string;
    icon?: string;
    className?: string;
  }
  ```
- **조합 패턴 적용**: props를 통한 커스터마이징 가능한 컴포넌트 설계


## 🙂 To Reviewer

@nan0silver 
기획과 화면이 동일한지 확인 부탁드립니다! 

## 📋 To Do

- [ ] API 연동 시작

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * 판매자 센터 대시보드, 상품 및 공동구매 등록 폼, 사이드바, 통계 카드, 상태 배지 등 다양한 판매자용 컴포넌트와 레이아웃이 추가되었습니다.
  * 공통 EmptyPage, ActionMenu, 옵션 리스트, 검색/필터 바, 섹션 헤더, 퀵 액션 카드 등 재사용 가능한 UI 컴포넌트가 도입되었습니다.
  * "이벤트", "숏구", 판매자 관련 여러 페이지에 준비중 안내 페이지가 추가되었습니다.
  * 상품 및 공동구매 등록을 위한 동적 폼과 이미지 업로드, 옵션 관리, 카테고리 선택 기능이 구현되었습니다.

* **Style**
  * 판매자 대시보드 및 공통 UI를 위한 Tailwind 기반 스타일이 대폭 추가되었습니다.
  * 여러 페이지와 그리드 레이아웃의 반응형 최대 너비 및 컬럼 수가 조정되었습니다.

* **Chores**
  * 새로운 데이터, 컴포넌트, 훅, 스타일, 상수 파일이 추가되고, 중앙 집중식 인덱스 파일이 도입되어 import가 간편해졌습니다.
  * 일부 경로, 내비게이션 라벨, 아이콘이 수정되었습니다.

* **Bug Fixes**
  * 내비게이션에서 "숏구" 경로가 잘못된 '/short'에서 '/shorts'로 수정되었습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

1000hyehyang committed:
feat: 판매자 대시보드 '상품 등록' 및 '공구 등록'  UI 구현

1000hyehyang committed:
refactor: 판매자 대시보드 사용하지 않는 페이지 정리

1000hyehyang committed:
refactor: 하드코딩 된 mock 데이터 분리

1000hyehyang committed:
design: 사이드바 메뉴 버튼 호버 시 디자인 수정

1000hyehyang committed:
refactor: 사용하지 않는 dead code 수정

1000hyehyang committed:
refactor: 메인 홈 1280 이상 그리드 깨짐 수정 및 모든 페이지 연결

nan0silver reviewed (approved):
판매자 상품 등록, 공구 등록 페이지가 잘 구현되어 보입니다!
메인 페이지 수정하신것도 확인했습니다!
수고하셨습니다!👍

1000hyehyang committed:
refactor: 코드 품질 및 안정성 개선

- EmptyPage 컴포넌트: ReactNode 타입으로 확장하여 더 유연한 사용 지원
  - 아이콘 컴포넌트, 리치 텍스트, 조건부 렌더링 지원
  - 기존 문자열 사용법과 호환성 유지

- useImageUpload 훅: 타입 안정성 및 에러 처리 강화
  - null/undefined 체크 추가
  - 파일 타입, 크기, 개수 검증 시스템 구현
  - 에러 상태 관리 및 피드백 기능 추가
  - 커스터마이징 가능한 옵션 인터페이스 제공

- GroupBuyRegistration: 파일 업로드 보안 강화
  - 이미지 파일 검증 로직 추가 (5MB 제한, 지원 형식 검증)
  - 파일 개수 제한 (최대 10개)
  - 명확한 에러 메시지 및 사용자 피드백
  - input 초기화로 잘못된 파일 제거

이번 수정으로 전반적인 코드 안정성과 사용자 경험이 크게 개선되었습니다.
